### PR TITLE
Refined Search By Adding Existing Stocks and Dropdown

### PR DESCRIPTION
**Pull request by javidangarcia**
- Added /stocks GET endpoint for getting all stocks that are currently in the database
- Added existing stocks to default search view using a table
- Added a dropdown to the search bar so that users can find stocks they are looking for

https://github.com/javidangarcia/stock-zone/assets/89328228/05c0f824-2f48-41d6-84f7-52631dde196f

### CODE REVIEW

File: frontend/src/components/Search/Search.css [hunk 1]
```diff
@@ -1,38 +1,108 @@
...
+}
+
+.search-table td,
+.search-table th {
+    text-align: center;
+    border: 1px solid #ddd;
+    padding: 8px;
+}
+
+.search-table tr:nth-child(even) {
```
Comment (vandanasub) [inline frontend/src/components/Search/Search.css:97]: cool!

**Review by vandanasub — COMMENTED** [review]
Nice! Dropdown for search looks great.

File: frontend/src/components/Search/Search.jsx [hunk 2]
```diff
@@ -1,29 +1,142 @@
...
+                <tbody>
+                    <tr>
+                        <th>Logo</th>
+                        <th>Ticker</th>
+                        <th>Name</th>
+                        <th>Sector</th>
+                        <th>Price</th>
+                    </tr>
+                </tbody>
+                {
```
Comment (vandanasub) [inline frontend/src/components/Search/Search.jsx:118]: an edge case: if db is empty, this component will only have the table headers. you could add some informative text for users in that case.

File: frontend/src/components/Search/Search.jsx [hunk 3]
```diff
@@ -1,29 +1,142 @@
...
+                                            </div>
+                                        </Link>
+                                    )
+                                }
+                            })
+                        }
+                    </div>
+                }
+            </div>
+            <table className="search-table">
```
Comment (vandanasub) [inline frontend/src/components/Search/Search.jsx:108]: if you expect this to get more complex in future, a good redesign is you could make the SearchTable its own component, SearchDropdown its own component, and Search as the parent component of both.

Comment (BriptimusPrimus) [reply frontend/src/components/Search/Search.jsx:108]: I would go as far as it is already complex enough that it should be split into separate components

**Comment by BriptimusPrimus** [comment]
Good job!

**Pull request merged**